src/FilterWrapper/components/FilterActionsPopover/FilterActionsPopover.tsx

```diff
@@ -1,102 +1,84 @@
-import * as React from "react";
 import cn from "classnames";
-import { OverflowVisibleContainer } from "../../../OverflowVisibleContainer";
+import * as React from "react";
 import { FilterAction } from "../../types";
 
 import "./FilterActionsPopover.scss";
 
-import {
-	IsntNeutralZoneMarker
-} from "../../../WindowClickListener";
+import { IsntNeutralZoneMarker } from "../../../WindowClickListener";
 
 interface FilterActionsPopoverProps {
-	filterActions: (FilterAction | any)[];
+	filterActions: Array<FilterAction | any>;
 	filterActionsCaption: string;
 }
 
-interface State {
-	isOpen: boolean;
-}
-
 const isFilterAction = (object: any): boolean => {
 	const filterAction = object as FilterAction;
-	return filterAction["key"] !== undefined
-		&& filterAction["onClick"] !== undefined
-		&& filterAction["name"] !== undefined;
-}
+	return (
+		filterAction.key !== undefined &&
+		filterAction.onClick !== undefined &&
+		filterAction.name !== undefined
+	);
+};
 
-export class FilterActionsPopover extends React.Component<
-	FilterActionsPopoverProps,
-	State
-	> {
-	public refElement = React.createRef<HTMLElement>();
+export const FilterActionsPopover: React.FC<FilterActionsPopoverProps> = ({
+	filterActions,
+	filterActionsCaption
+}) => {
+	const [isOpen, setIsOpen] = React.useState(false);
 
-	public state = {
-		isOpen: false
+	const handleClickFilter = () => {
+		setIsOpen(!isOpen);
```

Comment:
Вадим, возьми за правило менять стейт в зависимости от текущего через функцию
```suggestion
		setIsOpen(prevIsOpen => !prevIsOpen);
```